### PR TITLE
Log user safe exceptions

### PR DESCRIPTION
**Pull request by chenriksson**
Fix for: https://github.com/NuGet/Engineering/issues/809

### CODE REVIEW

**Comment by skofman1** [comment]
Are those logged as traces?

**Comment by skofman1** [comment]
In order to be able to monitor AI exceptions, we need AI exceptions to contain only exceptions that indicate a service issue. In this specific case, the exception is caused by a user mistake, and should be logged as a trace.

**Comment by chenriksson** [comment]
@skofman1 Will update PR to log as traces. Also need to investigate why user isn't getting the descriptive message (i.e., Id conflicts with Title of another package).

**Comment by chenriksson** [comment]
@skofman1 @joelverhagen Updated PR to send these exceptions to trace instead

File: src/NuGetGallery/Services/TelemetryService.cs [hunk 1]
```diff
@@ -36,6 +39,31 @@ public class TelemetryService : ITelemetryService
...
+        {
+            if (diagnosticsService == null)
+            {
+                throw new ArgumentNullException(nameof(diagnosticsService));
+            }
+
+            _trace = diagnosticsService.GetSource("TelemetryService");
+        }
+
+        // Used by ODataQueryVerifier. Should consider refactoring to make this non-static.
```
Comment (xavierdecoster) [inline src/NuGetGallery/Services/TelemetryService.cs]: Should we log an issue to track this as an actionable clean-up task?

Comment (chenriksson) [reply src/NuGetGallery/Services/TelemetryService.cs]: Good idea... opened #4795

**Pull request merged**